test/Interop/Cxx/objc-correctness/const-ref-parameter.swift

```diff
@@ -0,0 +1,77 @@
+
+// CHECK-IDE-TEST: struct OptionsConsumerCxx
+// CHECK-IDE-TEST: init(_ options: OptionsStruct)
+// CHECK-IDE-TEST: static func build(_ options: OptionsStruct) -> OptionsConsumerCxx
+// CHECK-IDE-TEST: static func doThing(_ options: OptionsStruct) -> Int32
+// CHECK-IDE-TEST: mutating func doOtherThing(_ options: OptionsStruct) -> Float
+
+
+// RUN: %target-swift-frontend -c -enable-experimental-cxx-interop -enable-objc-interop -I %S/Inputs %s -emit-sil -o - | %FileCheck %s
+
+// COM: FIXME: should it be @in_guaranteed OptionsStruct?
+// CHECK: [[FN1:%[0-9]+]] = function_ref @$sSo19OptionsConsumerObjCC7optionsABSo0A6StructV_tcfC : $@convention(method) (OptionsStruct, @thick OptionsConsumerObjC.Type) -> @owned OptionsConsumerObjC
+// CHECK-NEXT: apply [[FN1]]
+// CHECK-SAME: : $@convention(method) (OptionsStruct, @thick OptionsConsumerObjC.Type) -> @owned OptionsConsumerObjC
+
+// COM: FIXME: should it be @in_guaranteed OptionStruct?
+// CHECK: [[FN2:%[0-9]+]] = objc_method {{%[0-9]+}} : $OptionsConsumerObjC, #OptionsConsumerObjC.doOtherThing!foreign : (OptionsConsumerObjC) -> (OptionsStruct) -> Float, $@convention(objc_method) (@in OptionsStruct, OptionsConsumerObjC) -> Float
+// CHECK-NEXT: apply [[FN2]]
+// CHECK-SAME: : $@convention(objc_method) (@in OptionsStruct, OptionsConsumerObjC) -> Float
+
+// COM: FIXME: should it be @in_guaranteed OptionStruct?
+// CHECK: [[FN3:%[0-9]+]] = objc_method {{%[0-9]+}} : $@objc_metatype OptionsConsumerObjC.Type, #OptionsConsumerObjC.consumer!foreign : (OptionsConsumerObjC.Type) -> (OptionsStruct) -> @dynamic_self OptionsConsumerObjC, $@convention(objc_method) (@in OptionsStruct, @objc_metatype OptionsConsumerObjC.Type) -> @autoreleased OptionsConsumerObjC
+// CHECK-NEXT: apply [[FN3]]
+// CHECK-SAME: : $@convention(objc_method) (@in OptionsStruct, @objc_metatype OptionsConsumerObjC.Type) -> @autoreleased OptionsConsumerObjC
+
+// COM: FIXME: should it be @in_guaranteed OptionStruct?
+// CHECK: [[FN4:%[0-9]+]] = objc_method {{%[0-9]+}} : $@objc_metatype OptionsConsumerObjC.Type, #OptionsConsumerObjC.doThing!foreign : (OptionsConsumerObjC.Type) -> (OptionsStruct) -> Int32, $@convention(objc_method) (@in OptionsStruct, @objc_metatype OptionsConsumerObjC.Type) -> Int32
+// CHECK-NEXT: apply [[FN4]]
+// CHECK-SAME: : $@convention(objc_method) (@in OptionsStruct, @objc_metatype OptionsConsumerObjC.Type) -> Int32
+
+// COM: FIXME: should it be @in_guaranteed OptionStruct?
+// CHECK: [[FN5:%[0-9]+]] = function_ref @_ZN18OptionsConsumerCxxC1ERK13OptionsStruct : $@convention(c) (OptionsStruct) -> @out OptionsConsumerCxx
+// CHECK-NEXT: apply [[FN5]]
+// CHECK-SAME: : $@convention(c) (OptionsStruct) -> @out OptionsConsumerCxx
+
+// COM: FIXME: should it be @in_guaranteed OptionStruct?
+// CHECK: [[FN6:%[0-9]+]] = function_ref @_ZN18OptionsConsumerCxx12doOtherThingERK13OptionsStruct : $@convention(cxx_method) (@in OptionsStruct, @inout OptionsConsumerCxx) -> Float
```

Comment:
Is this passing on Windows?

Reply:
Oh, I see there's a requires objc at the top.

Reply:
I don't know what would change for Windows. The mangling?

As a later PR I can split the test into a pure C++ one and a Obj-C++ one, so at least part of the test runs in Linux and Windows. I don't have a Windows machine anymore, but I can check that Linux is consistent with Darwin.